File: ruby solutions/fibonacci.rb
class Sequences

  def fib_sequence(limit)
    base = [1, 2]
    while (base.last < limit)
      next_num = base[-1] + base[-2]
      base << next_num
    end
    base.pop
    base
  end

  def get_sum_evens(sequence)
    evens = sequence.keep_if {|num| num % 2 == 0 }
    evens.inject(:+)
  end

end


# Driver Code
# p (get_sum_evens(fib_sequence(4000000)) == 4613732) #=> true

File: javascript solutions/fibonacci.js


var fibSequence = function(limit) {
  var base = [1, 2];
  while (base.length < limit) {
    var next = base[base.length - 1] + base[base.length - 2];
    base.push(next);
  }

  console.log(base);
  return base;
}

var getSumEvens = function(sequence) {
  var result = [];
  var sum = 0;

  for (var i = 0; i < sequence.length; i++) {
    if (sequence[i] % 2 === 0) {
      result.push(sequence[i]);
    }
  }

  for (var j = 0; j < result.length; j++) {
    sum += result[j];
  }
  console.log(result);
  console.log(sum);
  return sum
}


var mySequence = fibSequence(10);
getSumEvens(mySequence);

var otherSeq = fibSequence(50);
getSumEvens(otherSeq);

